File: practices/day54-QRCodeNContactsNstaticlibs/day54-QRCodeNContactsNstaticlibs/src/cate/FileUtil.m
//
//  FileUtil.m
//  day35-yfcityhunt
//
//  Created by apple on 15/11/20.
//  Copyright (c) 2015年 yf. All rights reserved.
//

#import "FileUtil.h"

@implementation FileUtil

+(long long)fileSizeAtPath:(NSString *)path{
    BOOL b=0;
    if([iFm fileExistsAtPath:path isDirectory:&b]){
        if(!b)
            return [[iFm attributesOfItemAtPath:path error:0][NSFileSize] longLongValue];
        else{
            long long size=0;
           NSArray *subpaths= [iFm subpathsAtPath:path];
            for(NSString * file in subpaths){
                size+=[[iFm attributesOfItemAtPath:[path stringByAppendingPathComponent:file] error:0 ][NSFileSize] longLongValue];
            }
            return size;
        }
    }
    return 0;
}

+(NSString *)formatedFileSize2:(long long)size{
    NSString *strs[5]={@"B",@"K",@"M",@"G",@"T"};
    int idx=0;
    double resul=size;
    while (idx<4&&resul>1000) {
        if(idx==0)
            resul=size/1000.0;
        else
            resul=resul/1000.0;
        idx++;
    }
    if(idx==0)
        return [NSString stringWithFormat:@"%lld %@",size,strs[idx]];
    else
        return [NSString stringWithFormat:@"%.2f %@",resul,strs[idx]];
}


+(NSString *)formatedFileSize:(long long)size{
    NSString *strs[5]={@"B",@"K",@"M",@"G",@"T"};
    int idx=0;
    double resul=size;
    while (idx<4&&resul>1024) {
        if(idx==0)
            resul=size/1024.0;
        else
            resul=resul/1024.0;
        idx++;
    }
    if(idx==0)
        return [NSString stringWithFormat:@"%lld %@",size,strs[idx]];
    else
        return [NSString stringWithFormat:@"%.2f %@",resul,strs[idx]];
}

+(NSString *)cachePath{
    return  [NSSearchPathForDirectoriesInDomains(NSCachesDirectory, NSUserDomainMask, YES) firstObject];
}
+(NSString *)docPath{
    return [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject];
}
+(NSString *)tempPath{
    return NSTemporaryDirectory();
}

+(void)clearFileAtPath:(NSString *)path{
    if([iFm fileExistsAtPath:path]){
        NSArray *files=[iFm subpathsAtPath:path];
        for(NSString *file in files){
            NSString *absfile=[path stringByAppendingPathComponent:file];
            [iFm removeItemAtPath:absfile error:0];
        }
    }
}

@end

File: projects/EqApp/fjseism_ios/EqApp/src/ext/FileUtil.swift
//
//  FileUtil.swift
//  EqApp
//
//  Created by apple on 16/9/7.
//  Copyright © 2016年 apple. All rights reserved.
//

import Foundation
class FileUtil{
    
    class func fileSizeAtPath(path:String)->Int64{
        var b:ObjCBool=false
        if(iFm.fileExistsAtPath(path,isDirectory:&b)){
            if(b.boolValue){
                return try! iFm.attributesOfItemAtPath(path)[NSFileSize] as! Int64
            }
            else{
                var size:Int64=0
                let subpaths:[String] = iFm.subpathsAtPath(path) ?? []
                for file in subpaths{
                    size += try! iFm.attributesOfItemAtPath((path as NSString).stringByAppendingPathComponent(file))[NSFileSize] as! Int64
                    
                }
                return size;
            }
        }
        return 0;
    }
    
    class func formatedFileSize2(size:Int64)->String{
        let strs = ["B","K","M","G","T"]
        var idx:Int=0
        var resul:Double = Double(size)
        while (idx<4&&resul>1000) {
            if(idx==0){
                resul=Double(size)/1000.0;
            }
            else{
                resul=resul/1000.0;
            }
            idx += 1
        }
        if(idx==0){
            return String(format: "%lld %@", size,strs[idx])
            
        }
        else{
            return String(format: "%.2f %@", resul,strs[idx])
        }
    }
    
    
    class func formatedFileSize(size:Int64)->String{
        
        let strs = ["B","K","M","G","T"]
        var idx:Int=0
        var resul:Double = Double(size)
        while (idx<4&&resul>1024) {
            if(idx==0){
                resul=Double(size)/1024.0;
            }
            else{
                resul=resul/1024.0;
            }
            idx += 1
        }
        if(idx==0){
            return String(format: "%lld %@", size,strs[idx])
            
        }
        else{
            return String(format: "%.2f %@", resul,strs[idx])
        }
    }
    
    class func cachePath()->String{
        
        return  NSSearchPathForDirectoriesInDomains(.CachesDirectory, .UserDomainMask, true)[0]
    }
    class func docPath()->String{
        return NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)[0]
    }
    class func tempPath()->String{
        return NSTemporaryDirectory();
    }
    
    class func clearFileAtPath(path:String){
        if iFm.fileExistsAtPath(path){
            let files:[String] =  iFm.subpathsAtPath(path) ?? []
            for file in files{
                let apath = (path as NSString).stringByAppendingPathComponent(file)
                try! iFm.removeItemAtPath(apath)
            }
        }
    }
    
    
    class func newImgFile() ->String{
        return iConst.PNGDATESDF.stringFromDate(NSDate())
    
    }
    class func newVideoFile() ->String{
        return iConst.MP4DATESDF.stringFromDate(NSDate())
    }

}